google/cloud/storage/client.h

```diff
@@ -656,6 +656,30 @@ class Client {
     return ObjectWriteStream(raw_client_->WriteObject(request).second);
   }
 
+  /**
+   * Uploads a file to an object.
+   *
+   * @param file_name the name of the file to be uploaded.
+   * @param bucket_name the name of the bucket that contains the object.
+   * @param object_name the name of the object to be read.
+   * @param options a list of optional query parameters and/or request headers.
+   *   Valid types for this operation include `IfGenerationMatch`,
+   *   `IfGenerationNotMatch`, `IfMetagenerationMatch`,
+   *   `IfMetagenerationNotMatch`, `Generation`, and `UserProject`.
+   *
+   * @par Example
+   * @snippet storage_object_samples.cc upload file
+   */
+  template <typename... Options>
+  ObjectMetadata UploadFile(std::string const& file_name,
```

Comment:
I may have asked this already. I'm thinking upload RPC should have an optional Metadata parameter in case of a multi-part upload. This may be on the backlog, but I couldn't find the issue. I see multipart used implemented when md5 hash is provided, but it would be useful to extend to other metadata.

Most likely a follow-up action.

Reply:
Ack.  Yes, that should be a separate item (created #1311).